test/common/grpc/codec_test.cc

```diff
@@ -106,6 +106,59 @@ TEST(GrpcCodecTest, decodeInvalidFrame) {
   EXPECT_EQ(size, buffer.length());
 }
 
+// TODO: Should this test really pass? This test shows that null bytes in the bytestring
+// successfully decode into a frame with length 0.
+// A frame constructed from null bytes followed by an invalid frame
+TEST(GrpcCodecTest, DecodeMultipleFramesInvalid) {
+  const std::string data("\000\000\000\000\0000000", 9);
+  Buffer::OwnedImpl buffer(data.data(), data.size());
+
+  size_t size = buffer.length();
+
+  std::vector<Frame> frames;
+  Decoder decoder;
+  EXPECT_FALSE(decoder.decode(buffer, frames));
+  // When the decoder doesn't successfully decode, it puts decoded frames up until
+  // an invalid frame into output frame vector.
+  EXPECT_EQ(1, frames.size());
+  // Buffer does not get drained due to it returning false.
+  EXPECT_EQ(size, buffer.length());
+  // Only part of the buffer represented a frame. Thus, the frame length should not equal the buffer
+  // length.
+  EXPECT_NE(size, frames[0].length_);
```

Comment:
nice! thank you for the explanation, probably two checks on frame 0's length is unncessary, I would just choose the specific one below, but I love both the comments, keep them! :)